packages/muelu/src/Utils/MueLu_UtilitiesBase_decl.hpp

```diff
@@ -510,22 +510,47 @@ namespace MueLu {
      }
 #endif
 
-    /*! @brief Simple transpose for Tpetra::CrsMatrix types
+    /*! @brief Power method.
 
-        Note:  This is very inefficient, as it inserts one entry at a time.
+    @param A matrix
+    @param scaleByDiag if true, estimate the largest eigenvalue of \f$ D^; A \f$.
+    @param niters maximum number of iterations
+    @param tolerance stopping tolerance
+    @verbose if true, print iteration information
+    @seed  seed for randomizing initial guess
+
+    (Shamelessly grabbed from tpetra/examples.)
     */
+    static Scalar PowerMethod(const Matrix& A, bool scaleByDiag = true,
+                              LocalOrdinal niters = 10, Magnitude tolerance = 1e-2, bool verbose = false, unsigned int seed = 123) {
+      TEUCHOS_TEST_FOR_EXCEPTION(!(A.getRangeMap()->isSameAs(*(A.getDomainMap()))), Exceptions::Incompatible,
+          "Utils::PowerMethod: operator must have domain and range maps that are equivalent.");
+
+      // power iteration
+      RCP<Vector> diagInvVec;
+      if (scaleByDiag) {
+        RCP<Vector> diagVec = Xpetra::VectorFactory<Scalar,LocalOrdinal,GlobalOrdinal,Node>::Build(A.getRowMap());
+        A.getLocalDiagCopy(*diagVec);
+        diagInvVec = Xpetra::VectorFactory<Scalar,LocalOrdinal,GlobalOrdinal,Node>::Build(A.getRowMap());
+        diagInvVec->reciprocal(*diagVec);
+      }
+
+      Scalar lambda = PowerMethod(A, diagInvVec, niters, tolerance, verbose, seed);
+      return lambda;
+    }
 
     /*! @brief Power method.
 
     @param A matrix
-    @param scaleByDiag if true, estimate the largest eigenvalue of \f$ D^; A \f$.
+    @param invDiag reciprocal of matrix diagonal
     @param niters maximum number of iterations
```

Comment:
@jhux2 It looks like the parameter is now called diagInvVec .... while the above @param comment refers to invDiag